ios/EdgeCoreWebView.swift

```diff
@@ -0,0 +1,232 @@
+
+    // Set up the WKWebView child:
+    let webView = WKWebView(frame: bounds, configuration: configuration)
+    webView.navigationDelegate = self
+    addSubview(webView)
+    self.webView = webView
+
+    // Launch the core:
+    visitPage()
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+    webView?.frame = bounds
+  }
+
+  // callbacks -------------------------------------------------------------
+
+  func webView(
+    _: WKWebView,
+    didFailProvisionalNavigation: WKNavigation!,
+    withError: Error
+  ) {
+    // This makes webpack live-reload work correctly:
+    visitPage()
+  }
+
+  func webViewWebContentProcessDidTerminate(_: WKWebView) {
+    // Reload if we run out of memory:
+    visitPage()
+  }
+
+  func userContentController(
+    _: WKUserContentController,
+    didReceive scriptMessage: WKScriptMessage
+  ) {
+    if let call = scriptMessage.body as? NSArray,
+      let id = call[0] as? Int,
+      let name = call[1] as? String,
+      let args = call[2] as? NSArray
+    {
+      if id == 0 { return handleMessage(name, args: args) }
+
+      let promise = PendingCall(
+        resolve: { result in
+          self.runJs(
+            js: "window.nativeBridge.resolve(\(id), \(self.stringify(result)))")
+        },
+        reject: { message in
+          self.runJs(
+            js: "window.nativeBridge.reject(\(id), \(self.stringify(message)))")
+        })
+
+      do {
+        try handleCall(name, args: args, promise: promise)
+      } catch {
+        promise.reject("\(error)")
+      }
+    }
+  }
+
+  // utilities -------------------------------------------------------------
+
+  struct PendingCall {
+    var resolve: (_ value: Any?) -> Void
+    var reject: (_ message: String) -> Void
+  }
+
+  func handleCall(
+    _ name: String,
+    args: NSArray,
+    promise: PendingCall
+  ) throws {
+    if name == "diskletDelete", let path = args[0] as? String {
+      try disklet.delete(path: path)
+      return promise.resolve(nil)
+    }
+    if name == "diskletGetData", let path = args[0] as? String {
+      return promise.resolve(try disklet.getData(path: path).base64EncodedString())
+    }
+    if name == "diskletGetText", let path = args[0] as? String {
+      return promise.resolve(try disklet.getText(path: path))
+    }
+    if name == "diskletList", let path = args[0] as? String {
+      return promise.resolve(try disklet.list(path: path) as NSDictionary)
+    }
+    if name == "diskletSetData",
+      let path = args[0] as? String,
+      let base64 = args[1] as? String,
+      let data = Data.init(base64Encoded: base64)
+    {
+      try disklet.setData(path: path, data: data)
+      return promise.resolve(nil)
+    }
+    if name == "diskletSetText",
+      let path = args[0] as? String,
+      let text = args[1] as? String
+    {
+      try disklet.setText(path: path, text: text)
+      return promise.resolve(nil)
+    }
```

Comment:
All of the Disklet methods appear to run on the main thread and would block UI rendering. In contrast, scrypt is clearly run on a separate thread/queue. This is clearly shown when debugging the app and hitting a breakpoint in Disklet code which is on Thread 1. Fix to run on separate thread/queue.

<img width="823" alt="Screen Shot 2022-01-09 at 08 28 32" src="https://user-images.githubusercontent.com/6079354/148691498-b32139fe-7812-4d5c-a956-d72a0b85943f.png">
.

Reply:
These methods are fast, though. How much overhead is there in spinning up a thread, compared to just reading the data? I guess GCD makes the threading cheap & easy enough to be worthwhile, but I don't know if it makes as much sense on Android.

Reply:
The routines must be taking up enough of the main thread CPU time because I looked into it due to noticeable stuttering when comparing paneer versus develop . It was very slight but I am also on pretty fast phones. I imagine it would be significantly worse on older phones. 

I did only test on iOS and android might be a different story although I think it would be worth also putting that on another thread.

Reply:
Ok, I have made the change as a fixup commit, and have verified that the disklet code is running on another thread using the Xcode debugger.

For Android, the Disklet code already runs on a thread controlled by the WebView, so we don't need to do anything special there - it should not affect the GUI.